src/Controller/Api/DatasetPublicationController.php

```diff
@@ -258,19 +259,22 @@ public function linkAction(int $id, Request $request, ObjectPersister $objectPer
      *     )
      * )
      *
-    *
-    * @Route(
-    *     "/api/dataset_publications/{id}",
-    *     name="pelagos_api_dataset_publications_delete",
-    *     methods={"DELETE"},
-    *     defaults={"_format"="json"}
-    *     )
-    *
-    * @return Response A response object with an empty body and a "no content" status code.
-    */
-    public function deleteAction(int $id)
+     *
+     * @Route(
+     *     "/api/dataset_publications/{id}",
+     *     name="pelagos_api_dataset_publications_delete",
+     *     methods={"DELETE"},
+     *     defaults={"_format"="json"}
+     *     )
+     *
+     */
+    public function deleteAction(int $id, RabbitPublisher $publisher)
     {
+        $datasetPublication = $this->handleGetOne(DatasetPublication::class, $id);
+        $dataset = $datasetPublication->getDataset();
         $this->handleDelete(DatasetPublication::class, $id);
```

Comment:
handleDelete returns the entity which is just deleted.

Reply:
Should this instead be catching the possible "not deletable" BadRequestHttpException in its own try/catch block so it won't attempt to publish if the delete didn't happen?

Reply:
Even if the delete doesn't happen, the publish wouldnt unlink the publication doi from the datacite doi